File: app/src/main/java/com/ldp/reader/model/objectbox/ObjectBoxDbHelper.java
package com.ldp.reader.model.objectbox;

import com.ldp.reader.App;

import io.objectbox.BoxStore;

public class ObjectBoxDbHelper {
    private static volatile ObjectBoxDbHelper sInstance;

    private final BoxStore store;

    private ObjectBoxDbHelper() {
        store = MyObjectBox.builder()
                .androidContext(App.getContext())
                .build();
    }

    public static ObjectBoxDbHelper getInstance() {
        if (sInstance == null) {
            synchronized (ObjectBoxDbHelper.class) {
                if (sInstance == null) {
                    sInstance = new ObjectBoxDbHelper();
                }
            }
        }
        return sInstance;
    }

    public BoxStore getStore() {
        return store;
    }
}

File: app/src/main/java/com/ldp/reader/model/objectbox/ObjectBoxDbHelper.kt
package com.ldp.reader.model.objectbox

import com.ldp.reader.App
import io.objectbox.BoxStore

class ObjectBoxDbHelper private constructor() {
    val store: BoxStore = MyObjectBox.builder()
        .androidContext(App.getContext())
        .build()

    companion object {
        @Volatile
        private var sInstance: ObjectBoxDbHelper? = null

        @JvmStatic
        fun getInstance(): ObjectBoxDbHelper {
            if (sInstance == null) {
                synchronized(ObjectBoxDbHelper::class.java) {
                    if (sInstance == null) {
                        sInstance = ObjectBoxDbHelper()
                    }
                }
            }
            return sInstance!!
        }
    }
}
